File: ProjectJava/src/main/java/com/example/projectjava/Repository/UserDatabaseRepository.java
package com.example.projectjava.Repository;



import com.example.projectjava.Domain.*;
import com.example.projectjava.InterfaceRepositories.UserRepository;
import org.apache.logging.log4j.LogManager;
import org.apache.logging.log4j.Logger;

import java.sql.Connection;
import java.sql.PreparedStatement;
import java.sql.ResultSet;
import java.sql.SQLException;
import java.util.ArrayList;
import java.util.List;
import java.util.Properties;

public class UserDatabaseRepository implements UserRepository {
    private final JdbcUtils dbUtils;
    private static final Logger logger= LogManager.getLogger();

    public UserDatabaseRepository(Properties properties) {
        logger.info("Initializing UserDatabaseRepository with properties: {}",properties);
        dbUtils = new JdbcUtils(properties);
    }


    @Override
    public Iterable<User> getAll() {
        logger.traceEntry();

        Connection connection = dbUtils.getConnection();
        List<User> usersList = new ArrayList<>();
        try(PreparedStatement preparedStatement = connection.prepareStatement("SELECT * FROM user;")){
            try(ResultSet result = preparedStatement.executeQuery()){
                while(result.next()){
                    Integer userId = result.getInt("id");
                    String email = result.getString("email");
                    String password = result.getString("password");
                    String name = result.getString("name");
                    String status = result.getString("status");
                    UserStatus status1;
                    if(status=="LOGGEDIN")
                        status1=UserStatus.LOGGEDIN;
                    else
                        status1=UserStatus.LOGGEDOUT;
                    User user = new User(email,password,name,status1);
                    user.setId(userId);
                    usersList.add(user);
                }
            }
        }catch (SQLException exception){
            logger.error(exception);
        }
        logger.traceExit();
        return usersList;
    }

    @Override
    public User getOne(Integer id) throws Exception {
        logger.traceEntry();

        Connection connection = dbUtils.getConnection();
        User  user=null;
        try(PreparedStatement preparedStatement = connection.prepareStatement("SELECT * FROM user WHERE user.id = " + id + ";")){
            try(ResultSet result = preparedStatement.executeQuery()){
                if(result.next()){
                    String email = result.getString("email");
                    String password = result.getString("password");
                    String name = result.getString("name");
                    String status = result.getString("status");
                    UserStatus status1;
                    if(status.equals("LOGGEDIN"))
                        status1=UserStatus.LOGGEDIN;
                    else
                        status1=UserStatus.LOGGEDOUT;
                    user = new User(email,password,name,status1);
                    user.setId(id);
                }
            }
        }catch (SQLException exception){
            logger.error(exception);
        }
        logger.traceExit();
        return user;
    }

    @Override
    public User add(User entity) throws Exception {
        logger.traceEntry();
        Connection connection= dbUtils.getConnection();
        try(PreparedStatement preparedStatement=connection.prepareStatement("INSERT INTO user(email,password,name,status) VALUES (?,?,?,?);")){
            preparedStatement.setString(1,entity.getEmail());
            preparedStatement.setString(2,entity.getPassword());
            preparedStatement.setString(3,entity.getName());
            preparedStatement.setString(4,entity.getStatus().toString());
            int result = preparedStatement.executeUpdate();
            logger.trace("Saved {} instances",result);
        }catch (SQLException exception){
            logger.error(exception);
        }
        logger.traceExit();
        return entity;
    }

    @Override
    public User remove(Integer id) throws Exception {
        logger.traceEntry("Removing user with id {}",id);
        User user = getOne(id);
        Connection connection= dbUtils.getConnection();
        try(PreparedStatement preparedStatement = connection.prepareStatement("DELETE FROM user WHERE user.id = " + id+ ";")){
            int result = preparedStatement.executeUpdate();
            logger.trace("Deleted {} instances",result);
        }catch (SQLException exception){
            logger.error(exception);
        }
        return user;
    }

    @Override
    public User update(Integer id, User newEntity) throws Exception {
        logger.traceEntry("Modifying user with id  {}",id);
        User oldUser = getOne(id);
        Connection connection = dbUtils.getConnection();
        try(PreparedStatement preparedStatement = connection.prepareStatement("UPDATE user SET " +
                "email = '" + newEntity.getEmail() +
                "',password = '"+ newEntity.getPassword()+
                "',name = '" + newEntity.getName() +
                "',status = '" + newEntity.getStatus().toString()+
                "'  WHERE user.id ='" + id + "';"
        )){
            int result = preparedStatement.executeUpdate();
            logger.trace("Modified {} instances",result);
        }catch (SQLException exception){
            logger.error(exception);
        }
        logger.traceExit();
        return oldUser;
    }

    @Override
    public User getOne(String username,String password) throws Exception {
        logger.traceEntry();

        Connection connection = dbUtils.getConnection();
        User user=null;
        try(PreparedStatement preparedStatement = connection.prepareStatement("SELECT * FROM user WHERE user.email = " + username +
                " AND user.password = "+  password +";")){
            try(ResultSet result = preparedStatement.executeQuery()){
                if(result.next()){
                    Integer id = result.getInt("id");
                    String name = result.getString("name");
                    String status = result.getString("status");
                    UserStatus status1;
                    if(status.equals("LOGGEDIN"))
                        status1=UserStatus.LOGGEDIN;
                    else
                        status1=UserStatus.LOGGEDOUT;
                    user = new User(username,password,name,status1);
                    user.setId(id);
                }
            }
        }catch (SQLException exception){
            logger.error(exception);
        }
        logger.traceExit();
        return user;
    }

    @Override
    public User setUserStatus(Integer id, UserStatus userStatus) throws Exception {
        logger.traceEntry("Modifying user status  {}",id);
        User oldUser = getOne(id);
        Connection connection = dbUtils.getConnection();
        try(PreparedStatement preparedStatement = connection.prepareStatement("UPDATE user SET status = '" + userStatus.toString()+
                "' WHERE user.id = '" + id + "';"
        )){
            int result = preparedStatement.executeUpdate();
            logger.trace("Modified {} instances",result);
        }catch (SQLException exception){
            logger.error(exception);
        }
        logger.traceExit();
        return oldUser;
    }
}

File: ProjectC#/WindowsFormsApp/WindowsFormsApp/Repository/UserDatabaseRepository.cs
using ConsoleApp.Domain;
using ConsoleApp.InterfaceRepository;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using log4net;
using System.Configuration;
using System.Data;
using System.Data.SQLite;
using Dapper;
using System.Data;
using System.Data.SqlClient;

namespace ConsoleApp.Repository
{
    public class UserDatabaseRepository : UserRepository
    {
        private static readonly ILog Log = LogManager.GetLogger("UserDatabaseRepository");

        public UserDatabaseRepository()
        {
            Log.Info("Creating UserDatabaseRepository");
        }


        public List<User> findAll()
        {
            Log.Info("Entering GetAll");
            var users = new List<User>();
            string connectionString = "Data Source=LAPTOP-ON333H5E\\SQLEXPRESS;Initial Catalog=charityCaseSQLDatabase;Integrated Security=True";
            using (SqlConnection connection = new SqlConnection(connectionString))
            {
                SqlCommand command = new SqlCommand();
                command.Connection = connection;
                connection.Open();
                command.CommandText = "select * from userr;";
                

                using (var dataReader = command.ExecuteReader())
                {
                    while (dataReader.Read())
                    {
                        var id = dataReader.GetInt32(dataReader.GetOrdinal("id"));
                        var email = dataReader.GetString(dataReader.GetOrdinal("email"));
                        var name = dataReader.GetString(dataReader.GetOrdinal("name"));
                        var password = dataReader.GetString(dataReader.GetOrdinal("password"));
                        var status = dataReader.GetString(dataReader.GetOrdinal("status"));
                        UserStatus userStatus;
                        if (status == UserStatus.LOGGEDIN.ToString())
                        {
                            userStatus = UserStatus.LOGGEDIN;
                        }
                        else
                        {
                            userStatus = UserStatus.LOGGEDOUT;
                        }
                        var user = new User(email, password, name, userStatus);
                        user.setId(id);
                        users.Add(user);
                    }
                }
            }
            Log.InfoFormat("Exiting GetAll with values {0}", users);
            return users;
        }

        public User findOne(int id)
        {
            Log.InfoFormat("Entering GetOne with value {0}", id);

            User user = null;
            string connectionString = "Data Source=LAPTOP-ON333H5E\\SQLEXPRESS;Initial Catalog=charityCaseSQLDatabase;Integrated Security=True";
            using (SqlConnection connection = new SqlConnection(connectionString))
            {
                SqlCommand command = new SqlCommand();
                command.Connection = connection;
                connection.Open();
                command.CommandText = "select * from userr where userr.id = " + id + ";";
                using (var dataReader = command.ExecuteReader())
                {
                    if (dataReader.Read())
                    {
                        var email = dataReader.GetString(dataReader.GetOrdinal("email"));
                        var name = dataReader.GetString(dataReader.GetOrdinal("name"));
                        var password = dataReader.GetString(dataReader.GetOrdinal("password"));
                        var status = dataReader.GetString(dataReader.GetOrdinal("status"));
                        UserStatus userStatus;
                        if (status == UserStatus.LOGGEDIN.ToString())
                        {
                            userStatus = UserStatus.LOGGEDIN;
                        }
                        else
                        {
                            userStatus = UserStatus.LOGGEDOUT;
                        }
                        user = new User(email, password, name, userStatus);
                        user.setId(id);
                    }
                }
            }
            return user;
        }

        public void save(User user)
        {
            Log.InfoFormat("Entering Add with value {0}", user);
            string connectionString = "Data Source=LAPTOP-ON333H5E\\SQLEXPRESS;Initial Catalog=charityCaseSQLDatabase;Integrated Security=True";
            using (SqlConnection connection = new SqlConnection(connectionString))
            {
                SqlCommand command = new SqlCommand();
                command.Connection = connection;
                connection.Open();
                command.CommandText =
                "INSERT INTO userr(email,password,name,status) VALUES (@email,@password,@name,@status);";

                var dataParameter = command.CreateParameter();
                dataParameter.ParameterName = "@email";
                dataParameter.Value = user.getEmail();
                command.Parameters.Add(dataParameter);

                dataParameter = command.CreateParameter();
                dataParameter.ParameterName = "@password";
                dataParameter.Value = user.getPassword();
                command.Parameters.Add(dataParameter);

                dataParameter = command.CreateParameter();
                dataParameter.ParameterName = "@name";
                dataParameter.Value = user.getName();
                command.Parameters.Add(dataParameter);

                dataParameter = command.CreateParameter();
                dataParameter.ParameterName = "@status";
                dataParameter.Value = user.getStatus().ToString();
                command.Parameters.Add(dataParameter);

                var result = command.ExecuteNonQuery();
            }

            Log.InfoFormat("Exiting getOne with value {0}", user);
        }

        public void update(int id, User newEntity)
        {
            Log.InfoFormat("Entering Modify with value {0}", id);

            var oldUser = findOne(id);
            string connectionString = "Data Source=LAPTOP-ON333H5E\\SQLEXPRESS;Initial Catalog=charityCaseSQLDatabase;Integrated Security=True";
            using (SqlConnection connection = new SqlConnection(connectionString))
            {
                SqlCommand command = new SqlCommand();
                command.Connection = connection;
                connection.Open();
                command.CommandText = "UPDATE userr SET " +
                "email = '" + newEntity.getEmail() +
                "',password = '" + newEntity.getPassword() +
                "',name = '" + newEntity.getName() +
                "',status = '" + newEntity.getStatus() +
                "'  WHERE userr.id ='" + id + "';";
                var result = command.ExecuteNonQuery();
            }

            Log.InfoFormat("Exiting Remove with value {0}", oldUser);
        }

        public void delete(int id)
        {
            Log.InfoFormat("Entering Remove with value {0}", id);

            var user = findOne(id);
            string connectionString = "Data Source=LAPTOP-ON333H5E\\SQLEXPRESS;Initial Catalog=charityCaseSQLDatabase;Integrated Security=True";
            using (SqlConnection connection = new SqlConnection(connectionString))
            {
                SqlCommand command = new SqlCommand();
                command.Connection = connection;
                connection.Open();
                command.CommandText = "DELETE FROM userr WHERE userr.id = " + id + ";";
                var result = command.ExecuteNonQuery();
            }

            Log.InfoFormat("Exiting Remove with value {0}", user);
        }

        public User getOne(string username, string password)
        {

            User user = null;
            string connectionString = "Data Source=LAPTOP-ON333H5E\\SQLEXPRESS;Initial Catalog=charityCaseSQLDatabase;Integrated Security=True";
            using (SqlConnection connection = new SqlConnection(connectionString))
            {
                SqlCommand command = new SqlCommand();
                command.Connection = connection;
                connection.Open();
                command.CommandText = "select * from userr where userr.email = " + username + "and userr.password = " + password + ";";
                using (var dataReader = command.ExecuteReader())
                {
                    if (dataReader.Read())
                    {
                        var id = dataReader.GetInt32(dataReader.GetOrdinal("id"));
                        var name = dataReader.GetString(dataReader.GetOrdinal("name"));
                        var status = dataReader.GetString(dataReader.GetOrdinal("status"));
                        UserStatus userStatus;
                        if (status == UserStatus.LOGGEDIN.ToString())
                        {
                            userStatus = UserStatus.LOGGEDIN;
                        }
                        else
                        {
                            userStatus = UserStatus.LOGGEDOUT;
                        }
                        user = new User(username, password, name, userStatus);
                        user.setId(id);
                    }
                }
            }
            return user;
        }

        public User setUserStatus(int id, UserStatus userStatus)
        {
            User user = findOne(id);

            string connectionString = "Data Source=LAPTOP-ON333H5E\\SQLEXPRESS;Initial Catalog=charityCaseSQLDatabase;Integrated Security=True";
            using (SqlConnection connection = new SqlConnection(connectionString))
            {
                SqlCommand command = new SqlCommand();
                command.Connection = connection;
                connection.Open();
                command.CommandText = "UPDATE userr SET " +
                "status = '" + userStatus.ToString() +
                "'  WHERE userr.id ='" + id + "';";
                var result = command.ExecuteNonQuery();
            }
            return user;
        }
    }
}
